Yoast/wordpress-seo
# Added a post exclude filter

## Summary

This PR can be summarized in the following changelog entry:

* Adds a filter `wpseo_exclude_from_sitemap_by_post_ids` for modifying the excluded posts for the sitemap.

## Test instructions

This PR can be tested by following these steps:

* Checkout trunk
* Set some 'existing' post ids that you want to exclude. ( Yoast SEO - XML Sitemaps - Excluded posts )
* Visit the sitemap and if everything works as it should work the excludes posts are gone.
* Checkout this branch and add a filter to exclude some extra posts:
```php
add_filter( 'wpseo_exclude_from_sitemap_by_post_ids', function( $post_ids ) {
   $post_ids[] = '' // Here you have to enter an existing post id;

   return $post_ids;
} );
```
* You might have to change the amount of the sitemap entries to refresh the cache.
* Refresh the sitemap and see the filtered post id is gone as well.
* Press merge and grab some coffee. Thanks!
## Quality assurance


Fixes #7658
